### DISCUSSION TITLE
Smile5 compatibility?

### DISCUSSION THREAD

**Discussion by Freezerohmatic** [snitzelweck92/homeassistant_alphaess_modbus_tcp#5] [Q&A] [answered]
Hi All,

So I have a Smile5 and according to the documentation it does support Modbus over TCP. I'm unable to get this working and port 502 isn't open. I have a static IP set on the inverter and have scanned all ports on it, none are open/responding.  
I haven't tried restarting yet. 

Modbus is enabled and is set to slave.
EMS version       1.02.68
BMS version       1.15
Inverter version  1.48

Has anyone else had issues and over come them? Am I missing something?

**Comment by sebastianmachon** [comment]
Is your Smile5 connected via LAN?
I read that Modbus is disabled via WLAN.

**Reply by Freezerohmatic** [reply]
Thankyou for the fast response.

Yes it is on LAN. But I do have a 10.x.x.x network. Maybe it needs to be 192? 

I've also noticed I can't turn off modbus.

**Reply by sebastianmachon** [reply]
I have a 192. network, but I don't think that this should be the problem.

**Reply by Freezerohmatic** [reply]
I changed it to a 192 network and the port is still closed. It wouldn't surprise me if that was the cause.

I might try restarting it tomorrow.

**Reply by Freezerohmatic** [reply]
Tried restarting with no luck, might not be compatibile dispite it being listed?

**Comment by snitzelweck92** [comment]
Hello together,

I'm using also a 192.168 address.
This is not the problem :) you have to be sure, that you have no wlan
active.
Wifi and LAN is not working at same time.
You can check this, when the wifi dongle is not connected.
But for the smile 5 i'm not sure if modbus via tcp is available.
Sorry.

Freezerohmatic ***@***.***> schrieb am Mi., 25. Okt. 2023,
15:11:

**Reply by Freezerohmatic** [reply]
It is listed in the compatability list, ref AlphaESS-Handbuch_SMILE_ModBus_RTU_TCP_V21.pdf
Now that doesn't mean it works.

It doesn't have a WiFi module, its only connected via ethernet.

**Comment by Freezerohmatic** [comment, marked as answer]
Hi All,

I gave up on trying to get the TCP modbus working, I ended up modifying integration_alpha_ess.yaml to read off a RS485 USB device. Works perfectly, now I have a cable to manage but it works. 

This system is "old", being a few years old. I find it interesting that my inverter is listed to be compatible but does not seem to work. Either way I have a working solution. I'm happy to share the changes I made if anyone is interesed, maybe in a fork?

**Comment by snitzelweck92** [comment]
Yes you can :) i will see how to integrate

Freezerohmatic ***@***.***> schrieb am Do., 26. Okt. 2023,
11:52:

**Comment by Freezerohmatic** [comment]
Here are the changes I made. 

Changed 

```
  - name: modbuspvsystem
    type: tcp
    host: !secret alphaess_modbus_host_ip
    port: !secret alphaess_modbus_host_port
    message_wait_milliseconds: 5 # waittime between 2 messages
    timeout: 10 # timeout in seconds before connection is closed
    delay: 1 # delay in seconds at startup
```
  
To

```
- name: modbuspvsystem
    type: serial
    port: /dev/ttyUSB0 #USB FTDI device connected the the CAN/RS485 Modbus port on inverter
    baudrate: 9600
    bytesize: 8
    method: rtu
    parity: N
    stopbits: 1
    message_wait_milliseconds: 5 # waittime between 2 messages
    timeout: 10 # timeout in seconds before connection is closed
    delay: 1 # delay in seconds at startup
```

Thats all I had to do

**Reply by Freezerohmatic** [reply]
9600 baud is the default. Can be changed via modbus command....
